### ISSUE TITLE
Can you share perplexity during pretraining for some experiments 

### ISSUE THREAD

**Issue by ngoyal2707** [google-research/text-to-text-transfer-transformer#6]
Hi, first of all, great work, congrats!!! The experiments in the paper are very detailed and they help in answering some very interesting questions.

Can you please share pretraining perplexity of some experiments (similar to [BERT](https://arxiv.org/pdf/1810.04805.pdf) paper's Table 6). It will provide more understanding into these models. 

Specifically I am looking for:
1)  Final ppl achieved by models in `table 14` of your paper, ie. `T5-11B, T5-3B, T5-Large, T5-Base, T5-Small`.

2) Final ppl for experiments in `Table 13`.  

Thanks, I appreciate your time in answering these questions.

**Comment by craffel** [comment]
Thanks!

Do you want the train perplexity or validation perplexity? And do you want it for pre-training or fine-tuning? 

I'm curious why you want this. For Table 14, the perplexity will be averaged over both the unsupervised task and for the supervised tasks which will be on a totally different scale since for some of the supervised tasks (classification) it's trivial for the model to reduce the perplexity to small values (given the task prefix only ~two tokens are really valid).

**Comment by ngoyal2707** [comment]
@craffel Thanks for quick reply.
Ideally both `train_ppl` and `valid_ppl` of pre-training. Agree that fine-tuning ppl is not much interesting.

In terms of how this can be helpful, since `ppl` is decent measure of model quality during pretraining, I am curious about whether `(4xtraining, 1xmodel)` achieves lower ppl than `(1xtraining, 4xmodel)` etc. Also if the train/valid ppl of pretraining correlates highly to downstream task performance.
Similar for big models, I am curious as how much `T5-11b` does better on pretraining task compared to say `T5-large`.

Let me know if above make sense.

**Comment by craffel** [comment]
We don't monitor the validation perplexity. C4 is so big that none of our models could possibly overfit to it. For all of the experiments in the paper except those in Table 14 we don't even come close to making a single pass over the dataset; for the experiments in Table 14 we pre-train for 1T tokens which does end up being a single pass-ish depending on the mixing rate. But, overall, there's no substantive difference between train and validation perplexity for any of the experiments (except those in Figure 6/Table 9, where we artificially limit the size of C4, but that's the only case where we actually show the training loss).

For Table 14, the train perplexities are not comparable across models because they use different mixing rates (corresponding to a different artificial dataset size for each model). As a result some models see more or less unlabeled data, and as I mentioned above the perplexity for supervised tasks can be tiny due to the effectively limited set of reasonable tokens for classification tasks. I don't think you would learn anything from looking at those perplexities.

For Table 13, here is a plot of the training perplexities for the variants in the table (not including ensembles, since ensembling is done post-hoc):

![image](https://user-images.githubusercontent.com/417568/67983784-a3280380-fbe2-11e9-884d-26366913da02.png)

As a key, sc-bi_v1 is the baseline trained 4x as long, sc-bi_v1-bsx4 is the baseline trained with a 4x bigger batch size, sc-bi_v1-2x is the 2x larger model trained 2x as long, and sc-bi_v1-4x is the 4x bigger model trained for the same number of steps as the baseline.

**Comment by ngoyal2707** [comment]
This is very helpful, thanks for sharing!

**Comment by craffel** [comment]
Thanks for your interest!
